### PR TITLE
feat: voice state cache

### PR DESCRIPTION
**Pull request by fizvlad**
Caching voice state in `{guild ID => {user ID => voice state}}`

### CODE REVIEW

File: examples/voice_send.cr [hunk 1]
```diff
@@ -35,6 +37,17 @@ client.on_message_create do |payload|
...
+    # Used as:
+    # !vs
+
+    reply = begin
+      vs = cache.resolve_voice_state(payload.guild_id.not_nil!, payload.author.id)
+      "You are connected to channel #{vs.channel_id} at guild #{vs.guild_id}"
+    rescue
+      "No voice state"
+    end
+    client.create_message(payload.channel_id, reply)
```
Comment (PixeLInc) [inline examples/voice_send.cr:50]: I'd rather we didn't throw an exception and rescue it like this.
Prefer if we just make `resolve_voice_state` nilable and then do something like so:
```cr
reply = if vs = cache.resolve_voice_state(...)
              "You are connected to channel ...."
            else
              "No voice state"
            end

client.create_message(payload.channel_id, reply)

Comment (fizvlad) [reply examples/voice_send.cr:50]: As far as I remember, all of `resolve_*` methods in the lib do not return `nil`

Comment (PixeLInc) [reply examples/voice_send.cr:50]: meh. That's fair, alright.
I may change that in a future PR.

**Review by PixeLInc — APPROVED** [review]

**Comment by PixeLInc** [comment]
Thanks!

**Pull request merged**